mcs/class/Mono.Debugger.Soft/Mono.Debugger.Soft/StackFrame.cs

```diff
@@ -39,6 +39,8 @@ public class StackFrame : Mirror
 					if (vm.Version.AtLeast (2, 38)) {
 						try {
 							domain = vm.GetDomain (vm.conn.StackFrame_GetDomain (thread.Id, Id));
+						} catch (InvalidStackFrameException) {
+							domain = Thread.Domain;
```

Comment:
This is just masking an underlying problem.

Reply:
I know, but it also makes the behavior consistent with vm versions < 2.38 when the stack frame is invalid.